### DISCUSSION TITLE
Cannot run app built with pkg 5.8.1

### DISCUSSION THREAD

**Discussion by wt0m** [axios/axios#5866] [Q&A]
### Describe the bug

Pkg 5.8.1 throws these warnings when building my app, which uses axios 1.3.5 and node 16.18.1 :

> pkg@5.8.1
> Fetching base Node.js binaries to PKG_CACHE_PATH
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/index.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/axios.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/utils.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/bind.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/Axios.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/mergeConfig.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/defaults/index.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/formDataToJSON.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/cancel/CanceledError.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/cancel/CancelToken.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/cancel/isCancel.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/env/data.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/toFormData.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/AxiosError.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/spread.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/isAxiosError.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/AxiosHeaders.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/HttpStatusCode.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/buildURL.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/InterceptorManager.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/dispatchRequest.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/buildFullPath.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/validator.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/defaults/transitional.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/toURLEncodedForm.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/platform/index.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/platform/node/classes/FormData.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/parseHeaders.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/AxiosURLSearchParams.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/transformData.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/adapters/adapters.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/isAbsoluteURL.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/combineURLs.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/platform/node/index.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/adapters/http.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/adapters/xhr.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/platform/node/classes/URLSearchParams.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/core/settle.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/fromDataURI.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/AxiosTransformStream.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/formDataToStream.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/readBlob.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/ZlibHeaderTransformStream.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/cookies.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/isURLSameOrigin.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/parseProtocol.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/speedometer.js
> Warning Failed to make bytecode node16-x64 for file /snapshot/ils_configurator/node_modules/axios/lib/helpers/throttle.js

The app builds, but when I run it I get the following error:
```
pkg/prelude/bootstrap.js:1872
      throw error;
      ^

Error: Cannot find module '/snapshot/my_app/node_modules/axios/dist/node/axios.cjs'
1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
```

### To Reproduce

Just import axios with `import axios from 'axios'` and use it in your app.

### Code snippet

```js
import axios from 'axios'

const items = axios.create({
    baseURL: 'api/items'
})
```

### Expected behavior

The app runs without throwing an error.
Downgrading to axios 0.27.2 works but it is not an acceptable solution.

### Axios Version

1.3.5

### Node.js Version

16.18.1

### OS

Kubuntu 22.04

### Additional Library Versions

```bash
pkg 5.8.1
```

**Comment by akin-tekeoglu** [comment]
Can you try this solution?

https://github.com/vercel/pkg/issues/1822#issuecomment-1435836223

**Comment by wt0m** [comment]
Thanks for the quick reply.
I tried it, unfortunately it produces the same warnings and my app does not run.
Here's what I added to my package.json file:
```
"pkg": {
    "assets": [
      "node_modules/axios/dist/node/axios.cjs"
    ]
  },
```

**Comment by wt0m** [comment]
Any update on this? Other packages are compatible with pkg, as well as axios 0.27.2, it would be great if the newer versions were also compatible.
The solution posted above does not work for me.

**Comment by burdeasa** [comment]
I'm also seeing this issue, and the above workaround of adding the asset did not work for me either.

**Comment by rezvanniazi** [comment]
use `ncc` first and make it executable with pkg

**Reply by fanxiaodan-ra** [reply]
Does it work? I tried it, but it's not work for me, thank you
